UI/importers/classic.cpp

```diff
@@ -0,0 +1,577 @@
+/******************************************************************************
+    Copyright (C) 2019 by Dillon Pentz <dillon@vodbox.io>
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include "importers.hpp"
+
+#include <QByteArray>
+
+using namespace std;
+using namespace json11;
+
+static bool source_name_exists(string name, Json::array sources)
+{
+	for (size_t i = 0; i < sources.size(); i++) {
+		Json source = sources[i];
+		if (name == source["name"].string_value())
+			return true;
+	}
+
+	return false;
+}
+
+#define translate_int(in_key, in, out_key, out, off) \
+	out[out_key] = in[in_key].int_value() + off;
+#define translate_string(in_key, in, out_key, out) out[out_key] = in[in_key];
+#define translate_double(in_key, in, out_key, out) \
+	translate_string(in_key, in, out_key, out);
+#define translate_bool(in_key, in, out_key, out) \
+	out[out_key] = in[in_key].int_value() == 1;
+
+static Json::object translate_scene_item(Json in, Json source)
```

Comment:
Should be something like:
```cpp
static Json::object translate_scene_item(const Json &in, const Json &source)
```
if you're not modifying that data. If you don't use a reference, those objects will be copied. If you're not modifying the data, then they should also be const.